engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbProcessState.java

```diff
@@ -55,7 +55,7 @@ public final class DbProcessState implements MutableProcessState {
 
   private final Map<DirectBuffer, Long2ObjectHashMap<DeployedProcess>>
       processesByProcessIdAndVersion = new HashMap<>();
-  private final Map<String, Long2ObjectHashMap<DeployedProcess>> processByTenantAndKey;
+  private final Map<String, Long2ObjectHashMap<DeployedProcess>> processByTenantAndKeyCache;
```

Comment:
@korthout honestly, this can be much improved with a simple refactoring. Now it's a nested map. We should really add a record here that contains the tenant + process definition key and use an instance of this record as key for the hashmap. It's a lot cleaner and removed some complexity in the code.

I did this for the descision state, but it appears I forgot to do it here.

Reply:
To clarify, I mean a Java record. Not a Zeebe record 😛

Reply:
Thanks @remcowesterhoud 🙇 I'll have a look there and, if I like it, retro-actively apply it here